src/pineko/cli/check.py

```diff
@@ -36,54 +39,43 @@ def sub_compatibility(grid_path, operator_path, xif):
+
+    REN = ScaleValue("renormalization scale variations", check.contains_ren)
+    FACT = ScaleValue("factorization scale variations", check.contains_fact)
+
+
 @subcommand.command("scvar")
 @click.argument("grid_path", metavar="PINEAPPL", type=click.Path(exists=True))
 @click.argument(
     "scale",
     metavar="SCALE",
-    type=click.Choice(["ren", "fact"]),
+    type=click.Choice(list(el.name for el in Scale), case_sensitive=False),
 )
 @click.argument("max_as_order", metavar="AS_ORDER", type=int)
 @click.argument("max_al_order", metavar="AL_ORDER", type=int)
 def sub_scvar(grid_path, scale, max_as_order, max_al_order):
     """Check if PineAPPL grid contains requested scale variations for the requested order."""
     grid = pineappl.grid.Grid.read(grid_path)
     grid.optimize()
-    if scale == "ren":
-        is_ren_as, is_ren_al = check.contains_ren(grid, max_as_order, max_al_order)
-        if is_ren_as:
-            rich.print(
-                "[green]Success:[/] grids contain renormalization scale variations for as"
-            )
-        else:
-            rich.print(
-                "[red]Error:[/] grids do not contain renormalization scale variations for as"
-            )
-        if is_ren_al:
-            rich.print(
-                "[green]Success:[/] grids contain renormalization scale variations for al"
-            )
-        else:
-            rich.print(
-                "[red]Error:[/] grids do not contain renormalization scale variations for al"
-            )
-    elif scale == "fact":
-        is_fact_as, is_fact_al = check.contains_fact(grid, max_as_order, max_al_order)
-        if is_fact_as:
-            rich.print(
-                "[green]Success:[/] grids contain factorization scale variations for as"
-            )
-        else:
-            rich.print(
-                "[red]Error:[/] grids do not contain factorization scale variations for as"
-            )
-        if is_fact_al:
-            rich.print(
-                "[green]Success:[/] grids contain factorization scale variations for al"
-            )
+    success = "[green]Success:[/] grids contain"
+    error = "[red]Error:[/] grids do not contain"
+    # Call the function
+    try:
+        is_sv_as, is_sv_al = Scale[scale].value.check(grid, max_as_order, max_al_order)
+    except KeyError:
+        raise ValueError("Scale variation to check can be one between ren and fact")
+    conditions = {" for as": is_sv_as, " for al": is_sv_al}
+    for cond in conditions.keys():
+        to_write = ""
+        if conditions[cond]:
+            to_write += success
         else:
-            rich.print(
-                "[red]Error:[/] grids do not contain factorization scale variations for al"
-            )
-    else:
-        raise ValueError("Scale variation to check can be one between xir and xif")
+            to_write += error
+        to_write += " " + Scale[scale].value.descr
+        to_write += cond
```

Comment:
That is here:
```suggestion
        to_write += f" for {coupling.name}"
```
I used the `coupling` variable because at this point it will be just the name of the coupling.